.github/workflows/ci.yml

```diff
@@ -16,85 +16,56 @@ on:
-        include:
-          - toxenv: test-cov
-            os: ubuntu-latest
-            python-version: '3.11'
-          - toxenv: test-numpy122
-            os: ubuntu-latest
-            python-version: '3.10'
-          - toxenv: test-numpy121
-            os: ubuntu-latest
-            python-version: '3.10'
-          - toxenv: test-numpy122
-            os: ubuntu-latest
-            python-version: '3.9'
-          - toxenv: test-numpy121
-            os: ubuntu-latest
-            python-version: '3.9'
-          - toxenv: test-numpy120
-            os: ubuntu-latest
-            python-version: '3.9'
-          - toxenv: test-numpy122
-            os: ubuntu-latest
-            python-version: '3.8'
-          - toxenv: test-numpy121
-            os: ubuntu-latest
-            python-version: '3.8'
-          - toxenv: test-numpy120
-            os: ubuntu-latest
-            python-version: '3.8'
-          - toxenv: test-opencv-xdist
-            os: ubuntu-latest
-            python-version: '3.x'
-          - toxenv: test-jwst-xdist-cov
-            os: ubuntu-latest
-            python-version: '3.x'
-          - toxenv: test-romancal-xdist-cov
-            os: ubuntu-latest
-            python-version: '3.x'
-    steps:
-      - uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-      - uses: actions/setup-python@v4
-        with:
-          python-version: ${{ matrix.python-version }}
-      - run: pip install tox
-      - run: tox -e ${{ matrix.toxenv }}
-      - if: ${{ contains(matrix.toxenv,'-cov') }}
-        uses: codecov/codecov-action@v3
-        with:
-          file: ./coverage.xml
-          flags: unit
-          fail_ci_if_error: true
-          token: ${{ secrets.CODECOV_TOKEN }}
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@v1
+    with:
+      envs: |
+        - linux: test-xdist
+          python-version: '3.8'
+        - linux: test-xdist
+          python-version: '3.9'
+        - linux: test-xdist
+          python-version: '3.10'
+        - linux: test-xdist
+          python-version: '3.11'
+        - macos: test-xdist
+          python-version: '3.11'
+        - linux: test-opencv-xdist
+        - linux: test-cov-xdist
+          coverage: 'codecov'
+  test_numpy:
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@v1
+    with:
+      envs: |
+        - linux: test-numpy120-xdist
+          python-version: '3.8'
+        - linux: test-numpy121-xdist
+          python-version: '3.8'
+        - linux: test-numpy122-xdist
+          python-version: '3.8'
+        - linux: test-numpy120-xdist
+          python-version: '3.9'
+        - linux: test-numpy121-xdist
+          python-version: '3.9'
+        - linux: test-numpy122-xdist
+          python-version: '3.9'
+        - linux: test-numpy121-xdist
+          python-version: '3.10'
+        - linux: test-numpy122-xdist
+          python-version: '3.10'
```

Comment:
This seems excessive, it is also incomplete as `numpy` is up to version `1.24` I think.

It the max `numpy` version will be pulled into these tests naturally by `pip` during the `tox` environment creations. I suggest just doing a `minimum_dependencies` job, which will pull in the oldest dependencies (including `numpy`).